flag.go

```diff
@@ -452,7 +454,11 @@ func (f *FlagSet) Set(name, value string) error {
 	if err != nil {
 		var flagName string
 		if flag.Shorthand != "" && flag.ShorthandDeprecated == "" {
-			flagName = fmt.Sprintf("-%s, --%s", flag.Shorthand, flag.Name)
+			if flag.shorthandOnly {
+				flagName = fmt.Sprintf("-%s", flag.Shorthand)
```

Comment:
This it technically just suppressing the long name for the flag.
Is there a setup where the flag _only_ gets a shorthand? Something where you wouldn't have to explicitly specify `""` for the flag name?

Reply:
My first pass was to keep the API as unchanged as possible.

Let me see what I can draft up using a dedicated function.

Reply:
@jharshman This specific section is just printing the error. I think you're probably referencing L#841. But yeah, you're more or less right; I base `shorthandOnly` on `name == ""`. I'll try and open another branch and PR, because I don't want to touch this one. This one works and has for years. I have applications compiled against it.

Reply:
Okay, I've started another branch that will take a more explicit approach. I'll post it soon, but it's a much larger diff.